src/UpdatedPackagesCollector.js

```diff
@@ -127,7 +127,19 @@ export default class UpdatedPackagesCollector {
 
   collectUpdates() {
     return this.packages.filter(pkg => {
-      return this.updatedPackages[pkg.name] || (this.flags.onlyExplicitUpdates ? false : this.dependents[pkg.name]) || this.flags.canary;
+      if (this.updatedPackages[pkg.name]) {
+        return true;
+      }
+
+      if (!this.flags.onlyExplicitUpdates && this.dependents[pkg.name]) {
+        return true;
+      }
+
+      if (this.flags.canary && !pkg.isPrivate()) {
+        return true;
```

Comment:
I'd be happy to, but I'm not entirely sure what you had in mind. I'd still like to use private modules with canary releases, but if I understand @edvinerikson's comment correctly, he's suggesting disabling canary if the repo contains private modules.

My use case for private modules is example apps. I explicitly don't want to publish my example code via npm, because installing examples would be kinda silly, but i want the example app's dependencies to update correctly.